Added code to handle the case where autoResize=true. In this mode th…

gechols opened the pull request:
…e image size is preserved and the center is maintained.

This may be considered changing the default behavior so I'd welcome a suggestion to add a configuration parameter to help preserve the default behavior.

I believe this inadvertently resolves issue https://github.com/openseadragon/openseadragon/issues/459

iangilman commented on src/viewer.js:
```diff
@@ -2835,6 +2836,28 @@ function updateOnce( viewer ) {
+            bounds.height += viewportDiff.y;
+            bounds.x -= (viewportDiff.x / 2);
+            bounds.y -= (viewportDiff.y / 2);
+            viewer.viewport.fitBounds(bounds, true);
```
This should probably be `fitBoundsWithConstraints` so you don't end up zoomed further in or out than the constraints allow.

iangilman commented:
Yes, I think we need to continue to have `autoResize:false` mean that we don't do anything (allowing the developer to roll their own). Let's add a new option. It could be called something like `autoResizePreservesImageSize`. 

Regarding #459, this would only fix it if the dev picks the option. It's possible it could be fixed for the other case by using `fitBoundsWithConstraints` at https://github.com/openseadragon/openseadragon/blob/master/src/viewer.js#L2942

Thanks for tackling this!

gechols commented:
I added the configuration parameter (preserveImageSizeOnResize) which now preserves the default behavior. I also changed the fitBounds to fitBoundsWithConstraints as requested. I've tested it and it still works correctly.

iangilman commented:
Looking good, except I was thinking this new option should work only when autoResize is true (which it is by default); sorry I wasn't clear about that. At any rate, I think that makes more sense logically, since it is a form of auto resizing. Also it means less work for the dev; they just have to turn this one flag on rather than turning it on and autoResize off. Sound fair?

iangilman commented on src/viewer.js:
```diff
@@ -408,6 +408,7 @@ $.Viewer = function( options ) {
             width:             this.navigatorWidth,
             height:            this.navigatorHeight,
             autoResize:        this.navigatorAutoResize,
+            preserveImageSizeOnResize: this.preserveImageSizeOnResize,
```
I don't think we need this, as the navigator handles its own resizing.

gechols commented:
Refactored way preserveImageSizeOnResize was implemented - now it requires autoResize=true.

iangilman commented on src/openseadragon.js:
```diff
@@ -987,6 +990,7 @@ window.OpenSeadragon = window.OpenSeadragon || function( options ){
             maxZoomPixelRatio:      1.1, //-> higher allows 'over zoom' into pixels
             pixelsPerWheelLine:     40,
             autoResize:             true,
+            preserveImageSizeOnResize: false, // requires autoResize=false
```
Please update the comment here (autoResize=true)

iangilman commented:
Awesome, looks great. Thanks for the revisions! Just one minor nit and we can land it :)

gechols commented:
I'm confused why this caused the tests to fail. I thought we were preserving existing functionality unless we turned on the new parameter. @iangilman - do you have some insight into these tests to help me know what to fix?

iangilman commented:
Hmm... @connerman experienced something similar (see the chat in Gitter). Perhaps it's a bogus failure? After all, all you changed since the passing build was a comment. 

I've restarted the build...it failed a second time, so I've restarted again...and this time it passed. 

I'm seeing similar errors on master, so it doesn't have to do with this patch. I've filed #674 for us to investigate.
